File: test/test/HelloWorld/HelloWorld.go
package HelloWorld

import "fmt"

type HelloWorld struct {
}

func NewHelloWorld() (rcvr *HelloWorld) {
	rcvr = &HelloWorld{}
	return
}
func HelloWorld_main() {
	fmt.Println("Hello World")
}

type inner struct {
}

func NewInner() (rcvr *inner) {
	rcvr = &inner{}
	return
}
func (rcvr *inner) Test() (int) {
	fmt.Println("test1")
	return 1
}
func (rcvr *inner) TestStringint(s string, a int) (string) {
	fmt.Println("test4")
	return "returntest4"
}
func (rcvr *inner) Testint(a int) {
	fmt.Println("test2")
}
func (rcvr *inner) TestintString(a int, s string) (string) {
	fmt.Println("test3")
	return "returntest3"
}
func inner_main() {
	o := NewInner()
	fmt.Println(o.Test())
	o.Testint(1)
	fmt.Println(o.TestintString(1, "test3"))
	fmt.Println(o.TestStringint("test4", 1))
}

File: test/test/HelloWorld.java
package test;

public class HelloWorld {
    public static void main(String[] args) {
        System.out.println("Hello World");
    }
}

private class inner {
    public inner(){}
    public int test(){
            System.out.println("test1");
            return 1;
    }

    public void test(int a){
            System.out.println("test2");
    }
    public String test(int a,String s){
            System.out.println("test3");
            return "returntest3";
    }
    public String test(String s,int a){
            System.out.println("test4");
            return "returntest4";
    }
    public static void main(String[] args){
        inner o = new inner();
        System.out.println(o.test());
        o.test(1);
        System.out.println(o.test(1,"test3"));
        System.out.println(o.test("test4",1));
    }
}